File: bankingtool.client/src/app/dto/request/bank-account/deposit-amount-request.ts
import { RequestId } from "../request-id";

export interface DepositAmountRequestObject {
  request: DepositAmountRequest;
}

export interface DepositAmountRequest {
  account: DepositAmountRequestAccount;
  transaction: DepositAmountRequestTransaction;
}

export interface DepositAmountRequestAccount extends RequestId {

}

export interface DepositAmountRequestTransaction {
  amount: number;
  description: string;
}

export const CreateDepositAmountRequestObject = (accountId: number, amount: number, description: string): DepositAmountRequestObject => {
  let model: DepositAmountRequestObject = {
    request: {
      account: {
        id: accountId,
      },
      transaction: {
        amount: amount,
        description: description
      }
    }
  }
  return model;
}

File: BankingTool.Model/Dto/RequestDto/BankAccount/DepositAmountRequest.cs
using BankingTool.Model.Dto.RequestDtos;

namespace BankingTool.Model.Dto.RequestDto.BankAccount
{
    public class DepositAmountRequestObject
    {
        public DepositAmountRequest Request { get; set; }
    }
    public class DepositAmountRequest
    {
        public DepositAmountRequestAccount Account { get; set; }
        public DepositAmountRequestTransaction Transaction { get; set; }
    }
    public class DepositAmountRequestAccount : RequestId
    {
    }
    public class DepositAmountRequestTransaction
    {
        public int Amount { get; set; }
        public string Description { get; set; }
    }
}
